File: Java/Tema6_Ejercicio5.java
/*
 * TEMA 6: CADENAS DE CARACTERES - EJERCICIO 5
 * ============================================
 * Escriba un programa que reemplace caracteres en una cadena.
 */

import java.util.Scanner;

public class Tema6_Ejercicio5 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite una cadena: ");
        String cadena = scanner.nextLine();
        
        System.out.print("Digite el carácter a reemplazar: ");
        String viejo = scanner.nextLine();
        
        System.out.print("Digite el carácter nuevo: ");
        String nuevo = scanner.nextLine();
        
        String cadenaModificada = cadena.replace(viejo, nuevo);
        
        System.out.println("\n=== REEMPLAZO DE CARACTERES ===");
        System.out.println("Original: '" + cadena + "'");
        System.out.println("Reemplazar: '" + viejo + "' por '" + nuevo + "'");
        System.out.println("Resultado: '" + cadenaModificada + "'");
        
        // Método replaceAll con expresiones regulares
        String conRegex = cadena.replaceAll("[aeiou]", "*");
        System.out.println("Reemplazar vocales: '" + conRegex + "'");
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * Digite una cadena: programacion
 * Digite el carácter a reemplazar: a
 * Digite el carácter nuevo: o
 * 
 * === REEMPLAZO DE CARACTERES ===
 * Original: 'programacion'
 * Reemplazar: 'a' por 'o'
 * Resultado: 'progromocoon'
 * Reemplazar vocales: 'pr*gr*m*c**n'
 */

File: JS/Tema6_Ejercicio5.js
/*
 * TEMA 6: CADENAS DE CARACTERES - EJERCICIO 5
 * ============================================
 * Reemplazar contenido en strings.
 * 
 * EXPLICACIÓN:
 * - replace(): reemplaza primera ocurrencia
 * - replaceAll(): reemplaza todas (con regex o string)
 * - Usando expresiones regulares
 */

const texto = "gato come pescado, el gato es feliz y el gato juega";

console.log("=== REEMPLAZO DE CONTENIDO ===");
console.log(`Original: "${texto}"`);

// replace (solo primera)
const reemplazado1 = texto.replace("gato", "perro");
console.log(`\nReplace (1° ocurrencia): "${reemplazado1}"`);

// replaceAll (todas las ocurrencias)
const reemplazado2 = texto.replaceAll("gato", "perro");
console.log(`ReplaceAll (todas): "${reemplazado2}"`);

// Con expresión regular (case-insensitive)
const reemplazado3 = texto.replace(/gato/gi, "FELINO");
console.log(`Con regex (case-insensitive): "${reemplazado3}"`);

// Múltiples reemplazos
function reemplazarMultiple(str, buscar, reemplazos) {
    return buscar.reduce((acc, b, i) => acc.replaceAll(b, reemplazos[i]), str);
}

const resultado = reemplazarMultiple(
    "El día es hermoso y el día trae alegría",
    ["día", "hermoso", "alegría"],
    ["momento", "perfecto", "felicidad"]
);
console.log(`\nMúltiples reemplazos: "${resultado}"`);

/*
 * SALIDA ESPERADA:
 * === REEMPLAZO DE CONTENIDO ===
 * Original: "gato come pescado, el gato es feliz y el gato juega"
 * 
 * Replace (1° ocurrencia): "perro come pescado, el gato es feliz y el gato juega"
 * ReplaceAll (todas): "perro come pescado, el perro es feliz y el perro juega"
 * Con regex (case-insensitive): "FELINO come pescado, el FELINO es feliz y el FELINO juega"
 * 
 * Múltiples reemplazos: "El momento es perfecto y el momento trae felicidad"
 */
